crates/bevy_render/src/render_resource/bind_group_entries.rs

```diff
@@ -0,0 +1,241 @@
+use bevy_utils::all_tuples_with_size;
+use wgpu::{BindGroupEntry, BindingResource};
+
+use super::{Sampler, TextureView};
+
+/// Helper for constructing bindgroups.
+///
+/// Allows constructing the descriptor's entries as:
+/// ```ignore
+/// render_device.create_bind_group(
+///     "my_bind_group",
+///     &my_layout,
+///     &BindGroupEntries::with_indexes((
+///         (2, &my_sampler),
+///         (3, my_uniform),
+///     )),
+/// );
+/// ```
+///
+/// instead of
+///
+/// ```ignore
+/// render_device.create_bind_group(
+///     "my_bind_group",
+///     &my_layout,
+///     &[
+///         BindGroupEntry {
+///             binding: 2,
+///             resource: BindingResource::Sampler(&my_sampler),
+///         },
+///         BindGroupEntry {
+///             binding: 3,
+///             resource: my_uniform,
+///         },
+///     ],
+/// );
+/// ```
+///
+/// or
+///
+/// ```ignore
+/// render_device.create_bind_group(
+///     "my_bind_group",
+///     &my_layout,
+///     &BindGroupEntries::sequential((
+///         &my_sampler,
+///         my_uniform,
+///     )),
+/// );
+/// ```
+///
+/// instead of
+///
+/// ```ignore
+/// render_device.create_bind_group(
+///     "my_bind_group",
+///     &my_layout,
+///     &[
+///         BindGroupEntry {
+///             binding: 0,
+///             resource: BindingResource::Sampler(&my_sampler),
+///         },
+///         BindGroupEntry {
+///             binding: 1,
+///             resource: my_uniform,
+///         },
+///     ],
+/// );
+/// ```
+
+pub struct BindGroupEntries<'b, const N: usize> {
+    entries: [BindGroupEntry<'b>; N],
+}
+
+impl<'b, const N: usize> BindGroupEntries<'b, N> {
+    #[inline]
+    pub fn sequential(resources: impl IntoBindingArray<'b, N>) -> Self {
+        let mut i = 0;
+        Self {
+            entries: resources.into_array().map(|resource| {
+                let binding = i;
+                i += 1;
+                BindGroupEntry { binding, resource }
+            }),
+        }
+    }
+
+    #[inline]
+    pub fn with_indexes(indexed_resources: impl IntoIndexedBindingArray<'b, N>) -> Self {
```

Comment:
```suggestion
    pub fn with_indicies(indexed_resources: impl IntoIndexedBindingArray<'b, N>) -> Self {
```

Reply:
wouldn't it be indices? But also, I did a quick google and it seems like indexes is also commonly used in less formal context.

Reply:
Typo, yeah :P. Should be indices.